### PR TITLE
Prototype - Create PRs directly from Markdoc Sandbox

### PR DESCRIPTION
**Pull request by matv-stripe**
For fun

You have to create a github token - https://github.com/settings/tokens/new?scopes=repo  and then you can run

`GITHUB_AUTO_PR_TOKEN=ghp_yourtokenhere npm run dev`

https://user-images.githubusercontent.com/55463370/178815983-f1669cfe-de89-44f0-8805-c94eb8b5d50c.mov

Would be better if "create pr" opened up the github flow

### CODE REVIEW

File: components/Document.js [hunk 1]
```diff
@@ -7,13 +7,49 @@ function EditPage({ source: initialDocument }) {
...
+
+    const files = {
+      [filename]: doc
+    };
+
+    console.log(files);
+
+    const body = JSON.stringify({ files });
+
+    const response = await fetch('/api/create-pr', {
```
Comment (mfix-stripe) [inline components/Document.js:25]: I feel like we should just call this:
```suggestion
    const response = await fetch('/api/pr', {
```

and `POST` to it.

File: components/EditPagePanel.js [hunk 2]
```diff
@@ -3,7 +3,49 @@ import React from 'react';
...
+const MARKDOC_CLIENT_ID = '3a8716adbaa6294da1ec';
+function CreatePRButton({ createPR }) {
+  const [loading, setLoading] = React.useState(false);
+
+  return (
+    <div>
+      <button
+        className="create-pr-btn"
+        onClick={async () => {
+          // TODO: oauth is so annoying. We have to auth this and then get the code back, pass that to the server, but
```
Comment (mfix-stripe) [inline components/EditPagePanel.js:15]: Can we make this comment more straight to the point?

File: package-lock.json [hunk 3]
```diff
@@ -1,3795 +1,7 @@
...
-      "version": "3.1.1",
-      "resolved": "https://registry.npmjs.org/yn/-/yn-3.1.1.tgz",
-      "integrity": "sha512-Ux4ygGWsu2c7isFWe8Yu1YluJmqVhxqK2cLXNQA5AcC3QfbGNpM7fu0Y8b/z16pXLnFxZYvWhd3fhBY9DLmC6Q==",
-      "dev": true,
-      "engines": {
-        "node": ">=6"
-      }
-    }
-  },
+  "lockfileVersion": 1,
```
Comment (mfix-stripe) [inline package-lock.json:4]: Need to keep this version at 2

File: pages/api/create-pr.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+const initGithubClient = (token) => {
+  const octokit = new MyOctokit({
+    auth: TOKEN
+  });
+
+  return octokit;
+};
+
+const octokit = initGithubClient(TOKEN);
```
Comment (mfix-stripe) [inline pages/api/create-pr.js:19]:
```suggestion
const octokit = new MyOctokit({auth: TOKEN});
```

**Review by mfix-stripe — COMMENTED** [review]
@matv-stripe ik this is WIP, so FFT ignore, but just wanted to share a couple suggestions.

**Comment by matv-stripe** [comment]
Yeah sorry @mfix-stripe , hacky WIP prototype. I think there is actually value in actually shipping this - would be a very cool contributor experience to be able to edit directly.

But, could not get the o auth piece working well in an 30 minute time box. Closing the PR for now and will save it for another time.

**Pull request closed without merging**